mlir/include/mlir/Dialect/MIOpen/utility/builderUtils.h

```diff
@@ -11,6 +11,12 @@ namespace miopen {
 // Utility function to emit constant zero op. Can return scalars or vectors.
 Value createZeroConstantOp(OpBuilder &b, Location loc, Type type);
 
+// Utility function to emit type conversion ops.
+Value createTypeConversionOp(OpBuilder &b, Location loc, Value source,
```

Comment:
You don't need `sourceType` - it's just `Type sourceType = source.getType()` in the function

I think I made that change when pulling that function out in transforming_for

Great minds think alike, it seems